File: gethub.rb
require 'net/http'
require 'uri'
require 'io/console'

puts "Enter in a github username:"
gh_username = gets.chomp
puts "Enter in the github user password (hidden):"
gh_password = STDIN.noecho(&:gets).chomp

uri = URI.parse('https://api.github.com/user')

http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true

req = Net::HTTP::Get.new(uri.request_uri)
req.basic_auth(gh_username, gh_password)

r = http.request(req)

puts "\n#{r.body}"

File: gethub.py
import sys
import requests
import getpass

gh_username = raw_input("Enter in a github username:\n")
#gh_password = raw_input("Enter in the github user password (hidden):\n")
gh_password = getpass.getpass("Enter in the github user password (hidden):\n")

url = 'https://api.github.com/user'
auth = (gh_username, gh_password)

r = requests.get(url, auth=auth)
print r.content
